libs/language-server/src/test/extension/lib/test-property-meta-inf.ts

```diff
@@ -0,0 +1,104 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { TextDecoder } from 'util';
+
+import {
+  BlockMetaInformation,
+  EvaluationContext,
+  IOType,
+  PrimitiveValuetypes,
+  STRING_TYPEGUARD,
+  evaluatePropertyValueExpression,
+  isRuntimeParameterLiteral,
+} from '../../../lib';
+
+export class TestPropertyMetaInformation extends BlockMetaInformation {
+  constructor() {
+    super(
+      // How the block type should be called:
+      'TestProperty',
+      // Property definitions:
+      {
+        textProperty: {
+          type: PrimitiveValuetypes.Text,
+          docs: {
+            description: 'Required text property.',
+          },
+        },
+        customValidationTextProperty: {
+          type: PrimitiveValuetypes.Text,
+          defaultValue: 'utf-8',
+          docs: {
+            description:
+              'Text property with custom validation and default value.',
+          },
+          validation: (property, context) => {
+            const propertyValue = property.value;
+            if (isRuntimeParameterLiteral(propertyValue)) {
+              // We currently ignore runtime parameters during validation.
+              return;
+            }
+
+            const encodingValue = evaluatePropertyValueExpression(
+              propertyValue,
+              new EvaluationContext(), // we don't know values of runtime parameters or variables at this point
+              STRING_TYPEGUARD,
+            );
+
+            try {
+              new TextDecoder(encodingValue);
+            } catch (error) {
+              context.accept('error', `Unknown encoding "${encodingValue}"`, {
+                node: propertyValue,
+              });
+            }
```

Comment:
The validation logic could be simplified, e.g.
```suggestion
            const value = evaluatePropertyValueExpression(
              propertyValue,
              new EvaluationContext(), // we don't know values of runtime parameters or variables at this point
              STRING_TYPEGUARD,
            );

            if (value !== 'valid') {
              context.accept('error', `Invalid value "${value}"`, {
                node: propertyValue,
              });
            }
```